app/models/notification.rb

```diff
@@ -32,11 +32,15 @@ class Notification < ApplicationRecord
   end
 
   def parent_event
-    EventStore::Repository.find_event(parent_event_id)
+    @parent_event ||= EventStore::Repository.find_event(parent_event_id)
```

Comment:
this parent_event thing is because we use an intermediary event afair, right, not the notification event directly being the event of the notification. idk why it was done that way